src/contract/mod.rs

```diff
@@ -451,19 +475,53 @@ impl Document {
 
     pub fn get_raw_for_document_type<'a>(
         &'a self,
-        key: &str,
+        key_path: &str,
         document_type: &DocumentType,
         owner_id: Option<&[u8]>,
     ) -> Result<Option<Vec<u8>>, Error> {
-        if key == "$ownerId" && owner_id.is_some() {
+        if key_path == "$ownerId" && owner_id.is_some() {
             Ok(Some(Vec::from(owner_id.unwrap())))
         } else {
-            match self.properties.get(key) {
+            match key_path {
+                "$id" => return Ok(Some(self.id.clone())),
+                "$ownerId" => return Ok(Some(self.owner_id.clone())),
+                _ => {}
+            }
+            let key_paths: Vec<&str> = key_path.split('.').collect::<Vec<&str>>();
+            let (key, rest_key_paths) = key_paths.split_first().ok_or_else(|| Error::CorruptedData(String::from(
+                "key must not be null when getting from document",
+            )))?;
+
+            fn get_value_at_path<'a>(value: &'a Value, key_paths: &'a [&str]) -> Result<Option<&'a Value>, Error> {
```

Comment:
looks like an utility function, should it be moved out this function's body to be reusable?